Sources/TensorFlow/Core/Tensor.swift

```diff
@@ -229,6 +230,8 @@ public extension Tensor {
     /// - Precondition: The number of scalars must equal the product of the dimensions of the shape.
     @inlinable
     init(shape: TensorShape, scalars: [Scalar]) {
+        precondition(scalars.count == shape.contiguousSize,
+            "The number of scalars does not equal the product of dimensions given in the shape.")
```

Comment:
Could you include the `scalars.count` and `shape.contiguousSize` in the error messages here and below?

Reply:
That's a great idea. I'll add those in!

Reply:
I also reordered the precondition values to match the order of the argument labels in the function definition.